src/dir_status.ml

```diff
@@ -13,17 +18,22 @@ module T = struct
                     * Stanza.t list Dir_with_dune.t
     (* Directory with [(include_subdirs x)] where [x] is not [no] *)
 
-    | Is_component_of_a_group_but_not_the_root of
-        Stanza.t list Dir_with_dune.t option
+    | Is_component_of_a_group_but_not_the_root of is_component_of_a_group_but_not_the_root
     (* Sub-directory of a [Group_root _] *)
 
   let to_sexp _ = Sexp.Atom "<dir-status is opaque>"
 end
 include T
 
+type where_in_group =
+  | Not_in_group
+  | In_group of int (* depth *)
+
 let is_standalone = function
```

Comment:
This name is now a little weird. How about something like `group` or `get_group`.

Reply:
> I'm not a fan of this `depth` argument. It seems a bit error prone and annoying to manually maintained. It's only used to get the group root from a sub dir, right? Why can't we have a `group_root : Path.t` field instead of it?

I was thinking the same thing, and in fact have basically written the change you propose, but that broke tests when done naively. I think the reason is that for `Path.t` you have to make a choice of source directory vs build directory and I made the wrong choice.

For `depth`, there is no choice to make and so no opportunity for getting it wrong in this way, so I chose to keep it. 

Put differently, depth is a "relative" path, while `group_root` must be an absolute path, and relative path in this context seems less brittle.

I'm on the fence about this so if given this argument you're not convinced I'll make the change (being more careful about the source/build directory this time).

Reply:
I take that back, this change is shaping up nicely. I'll push it in a few minutes.

Reply:
> I'm on the fence about this so if given this argument you're not convinced I'll make the change (being more careful about the source/build directory this time).

Yeah, I see what's the issue. We have these `File_tree.Dir.t`'s from `Dir_status.t` so that we can combine source files and build targets to get a combined set of sources to process.

But when we call `Dir_contents.get`, we always have a build, rather than a source dir. Therefore, it should still possible to make it work with just paths.

If you'd like to leave `depth` or don't have time to refactor this, what about making it abstract in `dir_status.ml`

```
module Depth : sig
  type t
  val get : t -> path -> path
end
```

That would keep all ugliness in 1 module and allow us to experiment with getting rid of it later.

Reply:
Pushed.